src/Analyzers/VisualBasic/Analyzers/RemoveUnnecessaryByVal/VisualBasicRemoveUnnecessaryByVal.vb

```diff
@@ -0,0 +1,46 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Collections.Immutable
+Imports Microsoft.CodeAnalysis.CodeStyle
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.RemoveUnnecessaryByVal
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Friend Class VisualBasicRemoveUnnecessaryByVal
+        Inherits AbstractCodeStyleDiagnosticAnalyzer
+
+        Private Shared ReadOnly s_descriptor As DiagnosticDescriptor = New DiagnosticDescriptor(
+            id:=IDEDiagnosticIds.RemoveUnnecessaryByValDiagnosticId,
+            title:="",
+            messageFormat:="",
+            category:="",
+            defaultSeverity:=DiagnosticSeverity.Hidden,
+            isEnabledByDefault:=True,
+            description:="",
+            helpLinkUri:="",
+            WellKnownDiagnosticTags.Unnecessary)
+
+        Public Overrides ReadOnly Property SupportedDiagnostics As ImmutableArray(Of DiagnosticDescriptor) = ImmutableArray.Create(s_descriptor)
+
+        Public Sub New()
+            MyBase.New(s_descriptor.Id, s_descriptor.Title, s_descriptor.MessageFormat)
+        End Sub
+
+        Protected Overrides Sub InitializeWorker(ByVal context As AnalysisContext)
```

Comment:
How funny putting `ByVal` in this analyzer 😆 
I didn't know how to test the analyzer in action, added it to see if the analyzer would work or not. But surely it didn't work because the project uses v3.0 which doesn't contain it. Then, I forgot to remove it.